linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/LinkerdAdmin.scala

```diff
@@ -96,6 +112,21 @@ object LinkerdAdmin {
     }
   }
 
+  def extractInterpreterTransformerHandlers(routers: Seq[Router]): Seq[Admin.Handler] = {
+    routers.flatMap(router =>
+      router.interpreter match {
+        case withNameTreeTransformer: WithNameTreeTransformer =>
```

Comment:
Even if you decide not to go with for comprehensions, you can reuse the extraction of the admin handlers and avoid repeating yourself if you do something like: 

```scala
  def adminHandlers(a: WithNameTreeTransformer): Seq[Admin.Handler] =
    a.transformers.flatMap(transformer =>
    transformer match {
      case withHandlers: Admin.WithHandlers => withHandlers.adminHandlers
      case _ => Nil
    })

  def extractTransformersNamerHandlers(namers: Seq[Namer]): Seq[Admin.Handler] = {
    namers.flatMap{
        case withNameTreeTransformer: WithNameTreeTransformer => adminHandlers(withNameTreeTransformer)
        case _ => Nil
      }
  }

  def extractInterpreterTransformerHandlers(routers: Seq[Router]): Seq[Admin.Handler] = {
    routers.flatMap(router =>
      router.interpreter match {
        case withNameTreeTransformer: WithNameTreeTransformer => adminHandlers(withNameTreeTransformer)
        case _ => Nil
      })
  }
```

Would that work?

Reply:
I've used collect, but this would also work, thanks